internal/cli/atlas/setup/setup_cmd.go

```diff
@@ -0,0 +1,96 @@
+// Copyright 2022 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package setup
+
+import (
+	"context"
+
+	"github.com/mongodb/mongocli/internal/cli/atlas/quickstart"
+	"github.com/mongodb/mongocli/internal/cli/auth"
+	"github.com/mongodb/mongocli/internal/config"
+	"github.com/mongodb/mongocli/internal/flag"
+	"github.com/mongodb/mongocli/internal/store"
+	"github.com/mongodb/mongocli/internal/usage"
+	"github.com/spf13/cobra"
+)
+
+type Opts struct {
+	quickstart.Opts
+	auth.RegisterOpts
```

Comment:
I'm a bit worried about directly embedding whole working commands (is a kind of relationship) this way instead of embedding behaviour (behaves like kind of relationship) so far we extract the common behaviour to a new struct and never reuse a `Run`, which right now this type of relationship has 3 of, the `Run` from `quickstart.Opts`, the `Run` from `auth.RegisterOpts` and a new Run from `setup.Opts` making this command confusing to support in the long term

Reply:
since you don't really need the logic right now, until the patterns are clear I'd recommend to remove this embedding all together

Reply:
Agreed, we should not have 3 `Run`. Should I decouple the flags into another struct and reuse only them? I wanted to have the flags stablished as part of this PR

Reply:
at this time with no clear pattern I say: [A little copying is better than a little dependency.](https://www.youtube.com/watch?v=PAAkCSZUG1c&t=9m28s)

Reply:
Thanks! Done that.